src/etc/inc/dyndns.class

```diff
@@ -1543,6 +1767,78 @@
 						$this->_debug($data);
 					}
 					break;
+				case 'dreamhost':
+				case 'dreamhost-v6':
+					$result = json_decode($data,true);
+					if ($this->_dnsVerboseLog) {
+						log_error(sprintf(gettext('_checkStatus() results: %1$s'), $data));
+					}
+					switch ($result['data']) {
+						case 'success':
+						case 'record_added':
+						case 'record_removed':
+							$status = $status_intro . $success_str . gettext("IP Address Changed Successfully!") . " (" . $this->_dnsIP . ")";
+							$successful_update = true;
+							break;
+						case 'no_record':
+							$status = $status_intro . $error_str . gettext("No record exists.");
+							break;
+						case 'no_type':
+							$status = $status_intro . $error_str . gettext("No type exists.");
+							break;
+						case 'no_value':
+							$status = $status_intro . $error_str . gettext("No value exists.");
+							break;
+						case 'no_such_record ':
```

Comment:
Lots of unnecessary repetition throughout this block. Very similar error messages could also likely be combined if they are effectively the same. Use multiple cases before the code to help with that, such as:

```
case 'no_record':
case 'no_such_record ':
case 'invalid_record':
	$status = $status_intro . $error_str . gettext("No valid record exists.");
	break;
```